Modules/_sqlite/blob.c

```diff
@@ -102,8 +101,8 @@ pysqlite_close_all_blobs(pysqlite_Connection *self)
 {
     for (int i = 0; i < PyList_GET_SIZE(self->blobs); i++) {
         PyObject *weakref = PyList_GET_ITEM(self->blobs, i);
-        PyObject *blob = _PyWeakref_GET_REF(weakref);
-        if (blob == NULL) {
+        PyObject *blob;
+        if (!PyWeakref_GetRef(weakref, &blob)) {
```

Comment:
The `_sqlite` and `_ssl` modules both use some combination of `_PyWeakref_GET_REF` and `_PyWeakref_IS_DEAD`. Those are both static inline functions that end up calling unexported functions (`_PyMutex_LockTimed` and `_Py_IncRefTotal`), so linking fails. I don't love the idea of exporting both, but I guess that's OK.

Reply:
Ok, in that case I think the `_PyWeakref_GET_REF` -> `PyWeakref_GetRef` is fine.

For `_PyWeakref_IS_DEAD`, I think either:

1) Remove the locking in `_PyWeakref_IS_DEAD`. We don't really it. The call is inherently racy: just because it's not dead when you call `_PyWeakref_IS_DEAD` doesn't mean it's still alive in the next instant.
2) Or add the exported `_PyWeakref_IsDead` function (defined in `pycore_weakref.h`) for `_sqlite/connection.c`